examples/MomentoLoadGen/Program.cs

```diff
@@ -237,77 +236,87 @@ private async Task IssueAsyncSetGet(SimpleCacheClient client, CsharpLoadGenerato
                 {
                     var lastPrintCount = Interlocked.Exchange(ref context.LastWorkerStatsPrintRequestCount, globalRequestCount);
                     if (lastPrintCount != globalRequestCount)
                     {
                         Console.WriteLine($"worker: {workerId} last print count: {lastPrintCount} global request count: {globalRequestCount}");
-                        _logger.LogInformation($"worker: {workerId}, worker request: {operationId}, global request: {context.GlobalRequestCount}, status: {getResult.GetType()}, val: {valueString}");
+                        _logger.LogInformation($"worker: {workerId}, worker request: {operationId}, global request: {context.GlobalRequestCount}, status: {getResponse.GetType()}, val: {valueString}");
                     }
                 }
             }
         }
 
-        private async Task<TResult?> ExecuteRequestAndUpdateContextCounts<TResult>(
+        private async Task<TResult> ExecuteRequestAndUpdateContextCounts<TResult>(
             CsharpLoadGeneratorContext context,
-            Func<Task<TResult>> block
+            Func<Task<Tuple<AsyncSetGetResult, TResult>>> block
             )
-        {
-            var result = await ExecuteRequest(block);
+        {            
+            var result = await block();
             UpdateContextCountsForRequest(context, result.Item1);
             return result.Item2;
         }
 
-
-        private async Task<Tuple<AsyncSetGetResult, T?>> ExecuteRequest<T>(
-            Func<Task<T>> block
-            )
+        private async Task<Tuple<AsyncSetGetResult, CacheGetResponse?>> IssueGetRequest(SimpleCacheClient client, String cacheKey)
         {
-            try
+            var getResponse = await client.GetAsync(CACHE_NAME, cacheKey);
+            if (getResponse is CacheGetResponse.Hit hit)
             {
-                T result = await block();
-                return Tuple.Create(AsyncSetGetResult.SUCCESS, (T?)result);
-            }
-            catch (InternalServerException e)
+                return Tuple.Create<AsyncSetGetResult, CacheGetResponse?>(AsyncSetGetResult.SUCCESS, hit);
+            } else if (getResponse is CacheGetResponse.Miss miss)
             {
-                var innerException = e.InnerException as RpcException;
-                _logger.LogWarning("CAUGHT AN EXCEPTION WHILE EXECUTING REQUEST: {0}", innerException);
-                switch (innerException!.StatusCode)
-                {
+                return Tuple.Create<AsyncSetGetResult, CacheGetResponse?>(AsyncSetGetResult.SUCCESS, miss);
+            } else if (getResponse is CacheGetResponse.Error error)
+            {
+                return Tuple.Create<AsyncSetGetResult, CacheGetResponse?>(ConvertErrorToAsyncSetGetResult(error.ErrorCode, error.Exception), null);
+            } else
+            {
+                throw new ApplicationException($"Unsupported get response: {getResponse}");
+            }
```

Comment:
total nit but I prefer switch statement over the multiline if. Not sure if c# reason though for doing in IF is cleaner with how we pass back error codes so not blocker at all. Just personal preference thing for me. Saw we had switch on status before.

Reply:
Unfortunately the blessed/idiomatic way to do this kind of pattern matching in C# today is with if statements:

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#type-tests

I would rather have it as a switch too, but the smart casting to the local variable is kind of a crucial part of the user experience here now.

Reply:
ack figured was something like this.